<a href="https://colab.research.google.com/github/DinaShaim/GB_ML_MathStat/blob/main/%D0%94%D0%BE%D0%BC%D0%B0%D1%88%D0%BD%D1%8F%D1%8F_%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%B0_%D0%9C%D0%B0%D1%82%D1%81%D1%82%D0%B0%D1%82_%D0%A3%D1%80%D0%BE%D0%BA_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Домашняя работа 5
### Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

**Задание 1** 

Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

**Решение**



Вычислим доверительный интервал по формуле:

$$T_{1,2} = \overline{X} \pm  Z_{a/2}\cdot \frac{σ}{\sqrt{n}}  $$

$$T_{1,2} = 80 \pm  1.96\cdot \frac{16}{16}  $$

Доверительнвй интерал равен: 

$$ [78.04; 81.96]$$

**Задание 2** 

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95. 

**Решение**

Обозначим известные данные:

In [ ]:
import numpy as np
import pandas as pd
df = pd.DataFrame({'values' : [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1] })
X = df['values'].mean()
X

6.589999999999999

In [ ]:
σ = df['values'].std(ddof=1)
σ

0.4508017549014448

Обьем выборки равен 10, значит степеней свободы будет 9 и значение критерия возьмем из таблицы, тогда:

Вычислим доверительный интервал по формуле:

$$T_{1,2} = \overline{X} \pm  t_{a/2}\cdot \frac{σ}{\sqrt{n}}  $$

In [ ]:
T1 = X - 2.262*(σ/np.sqrt(10))
T1

6.267538255912425

In [ ]:
T2 = X + 2.262*(σ/np.sqrt(10))
T2

6.912461744087573

Доверительный интервал равен [6.268 ; 6.912]

**Задание 3** 

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

**Решение**

Сформулируем гипотезы:

H0: $M = M0$

H1: $M > M0$

Поскольку известна дисперсия, то воспользуемся Z критерием. Расчетное $Z_{p}$ вычислим по формуле: 

$$Z_{p} = \frac{M - M0}{\sigma/\sqrt{n}}$$

In [ ]:
M0 = 17
M = 17.5
a = 5
n = 100
σ = np.sqrt(4)
Zp = (M - M0)/ (σ/np.sqrt(n))
Zp

2.5

Поскольку $α$ равно 5, то табличное значение расчитываем для 95%. 

In [ ]:
Zt = 1.645

Поскольку $Zp > Zt$, то отвергаем нулевую гипотезу на уровне значимости 5%.

**Задание 4**

Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

**Решение**

Обозначим все известные данные:

In [ ]:
df = pd.DataFrame({'values' : [202, 203, 199, 197, 195, 201, 200, 204, 194, 190] })
M0 = 200
a = 1
n = 10

Расчитаем математическое ожидание, средние квадратическое отклонение:

In [ ]:
M = df['values'].mean()
M

198.5

In [ ]:
σ = df['values'].std(ddof=1)
σ

4.453463071962462

Используем критерий стьюдента. Расчитаем $tp$:

In [ ]:
tp = (M - M0) / (σ/np.sqrt(n))
tp

-1.0651074037450896

Табличное значение критерия стьюдента расчитываем для $α/2 = 0.005$ и степени свободы 9. 

In [ ]:
tt = 3.25

Для отрицательных значений будет равно -3.25.

Поскольку $-1.065 > -3.25$, т.е. $tp > tt$, то H0 верна на уровне значимости 1%.